src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectlyTests.Fixer.cs

```diff
@@ -29,5 +30,198 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
         {
             return new CSharpDoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectlyFixer();
         }
+
+        [Fact]
+        public void CA1826FixEnumerableFirstExtensionCallCSharp()
```

Comment:
Can you please add a corresponding VisualBasic test? I think a single VisualBasic test should suffice given the fix code is completely in language agnostic layer.

Reply:
The fixer references the `InvocationExpressionSyntax` and `ExpressionSyntax` classes which come from `Microsoft.CodeAnalysis.CSharp.Syntax` namespace. I assume that makes it C# specific.

Do you want me to move those parts to the C# specific subclass? I can attempt to make it work for VB as well?

Reply:
Oh I just completely overlooked that - C# specific references should not be accessed in the common layer. That is a layering violation. I am fine with you taking either approach:
1. Move it to C# layer and make the current implementation C#-only, with possible future extension to VB.
2. Add VB/C# specific implementations in respective layers and add unit tests for VisualBasic.